pkg/infra/iac2/templates_compiler_test.go

```diff
@@ -21,13 +20,13 @@ func TestOutputBody(t *testing.T) {
 		Fizz: fizz,
 		Buzz: buzz,
 	}
-	graph := graph2.NewDirected[core.Resource]()
-	graph.AddVertex(fizz)
-	graph.AddVertex(buzz)
-	graph.AddVertex(parent)
-	graph.AddEdge(fizz.Id(), parent.Id())
-	graph.AddEdge(buzz.Id(), parent.Id())
-	graph.AddEdge(buzz.Id(), fizz.Id())
+	graph := core.NewResourceGraph()
+	graph.AddResource(fizz)
+	graph.AddResource(buzz)
+	graph.AddResource(parent)
+	graph.AddDependency(fizz, parent)
+	graph.AddDependency(buzz, parent)
+	graph.AddDependency(buzz, fizz)
```

Comment:
Shoot, I forgot to add a comment on this line. Since you're touching it anyway, mind adding this?

```suggestion
	graph.AddDependency(buzz, fizz) // provides a stable topological ordering
```